### ISSUE TITLE
RA reports missing trait item implementation for associated constant with default value: possibly a regression.

### ISSUE THREAD

**Issue by nerditation** [rust-lang/rust-analyzer#15909]
# description

a trait has an associated constant with a default value, when a type implements the trait and omit the associated constant, RA will show error message:

> "not all trait item implemented, missing: `const XXXX`"

while the same code compiles without any problem.

I can't remember exactly, but I think this error starts to show up from yesterday; in previous versions, no error is shown for the exact same code base.

minimal reproducible code:

```rust
trait Marker {
	const FLAG: bool = false;
}
struct Foo;
impl Marker for Foo {}  // <--- error message shows up here
```

## screenshot

![image](https://github.com/rust-lang/rust-analyzer/assets/12248559/76e795a9-6e1c-4fef-86d6-356be163da1e)

**rust-analyzer version**: 

using vscode extension preview channel,

- rust-analyzer 0.4.1735-standalone (58de0b130 2023-11-15)
- Extension version: 0.4.1735

```console
INFO [11/16/2023, 10:29:17 AM]: Starting language client
INFO [11/16/2023, 10:29:17 AM]: Using server binary at d:\Applications\scoop\apps\vscode\1.83.1\data\extensions\rust-lang.rust-analyzer-0.4.1735-win32-x64\server\rust-analyzer.exe
INFO [11/16/2023, 10:29:17 AM]: d:\Applications\scoop\apps\vscode\1.83.1\data\extensions\rust-lang.rust-analyzer-0.4.1735-win32-x64\server\rust-analyzer.exe --version: {
  status: 0,
  signal: null,
  output: [
    null,
    'rust-analyzer 0.4.1735-standalone (58de0b130 2023-11-15)\n',
    ''
  ],
  pid: 17980,
  stdout: 'rust-analyzer 0.4.1735-standalone (58de0b130 2023-11-15)\n',
  stderr: ''
}
```

**rustc version**:

```console
rustc 1.76.0-nightly (dd430bc8c 2023-11-14)
```

**relevant settings**: 

nothing special, just normal installation via rustup and vscode

**Comment by Young-Flash** [comment]
seems it is introduced in https://github.com/rust-lang/rust-analyzer/pull/15895, I'll take a look

**Comment by Young-Flash** [comment]
got the problem, will make a patched PR soon
